packages/command/src/sfdxFlags.ts

```diff
@@ -45,14 +45,25 @@ function toValidatorFn(validator?: unknown): (val: string) => boolean {
   };
 }
 
+function merge<T>(
+  kind: flags.Kind,
+  flag: IOptionFlag<T | undefined>,
+  describable: flags.Describable
+): flags.Discriminated<flags.Option<T>>;
+function merge<T>(
+  kind: flags.Kind,
+  flag: IBooleanFlag<T>,
+  describable: flags.Describable
+): flags.Discriminated<flags.Boolean<T>>;
 function merge<T>(kind: flags.Kind, flag: IFlag<T>, describable: flags.Describable): flags.Discriminated<flags.Any<T>> {
   if (has(flag, 'validate') && hasFunction(flag, 'parse')) {
-    const parse: <O>(val: string, ctx: unknown) => O = flag.parse.bind(flag);
-    flag.parse = <O>(val: string, ctx: unknown): O => {
+    const parse = flag.parse.bind(flag);
+    flag.parse = <T>(val: string, ctx: unknown): T => {
```

Comment:
Preserves the `T` but then cast to `T` in the return. I don't think having the intermediate `parse` typed with the type argument `<O>` did anything because `O` was never actually supplied.

Reply:
Ya, that seems fine.